libindy/src/services/wallet/mod.rs

```diff
@@ -816,139 +816,167 @@ mod tests {
 
     #[test]
     fn wallet_service_register_type_works() {
-        _cleanup();
+        _cleanup("wallet_service_register_type_works");
 
         let wallet_service = WalletService::new();
         _register_inmem_wallet(&wallet_service);
+
+        _cleanup("wallet_service_register_type_works");
     }
 
     #[test]
     fn wallet_service_create_wallet_works() {
-        _cleanup();
-
-        let wallet_service = WalletService::new();
-        wallet_service.create_wallet(&_config_default(), &RAW_CREDENTIAL, (&RAW_KDD, &RAW_MASTER_KEY)).unwrap();
+        test::cleanup_wallet("wallet_service_create_wallet_works");
+        {
+            let wallet_service = WalletService::new();
+            wallet_service.create_wallet(&_config_default("wallet_service_create_wallet_works"), &RAW_CREDENTIAL, (&RAW_KDD, &RAW_MASTER_KEY)).unwrap();
+        }
+        test::cleanup_wallet("wallet_service_create_wallet_works");
     }
 
     #[test]
     fn wallet_service_create_wallet_works_for_interactive_key_derivation() {
-        _cleanup();
-
-        let wallet_service = WalletService::new();
-        wallet_service.create_wallet(&_config_default(), &ARGON_INT_CREDENTIAL, (&INTERACTIVE_KDD, &INTERACTIVE_MASTER_KEY)).unwrap();
+        test::cleanup_wallet("wallet_service_create_wallet_works_for_interactive_key_derivation");
+        {
+            let wallet_service = WalletService::new();
+            wallet_service.create_wallet(&_config_default("wallet_service_create_wallet_works_for_interactive_key_derivation"), &ARGON_INT_CREDENTIAL, (&INTERACTIVE_KDD, &INTERACTIVE_MASTER_KEY)).unwrap();
+        }
+        test::cleanup_wallet("wallet_service_create_wallet_works_for_interactive_key_derivation");
     }
 
     #[test]
     fn wallet_service_create_wallet_works_for_moderate_key_derivation() {
-        _cleanup();
-
-        let wallet_service = WalletService::new();
-        wallet_service.create_wallet(&_config_default(), &ARGON_MOD_CREDENTIAL, (&MODERATE_KDD, &MODERATE_MASTER_KEY)).unwrap();
+        test::cleanup_wallet("wallet_service_create_wallet_works_for_moderate_key_derivation");
+        {
+            let wallet_service = WalletService::new();
+            wallet_service.create_wallet(&_config_default("wallet_service_create_wallet_works_for_moderate_key_derivation"), &ARGON_MOD_CREDENTIAL, (&MODERATE_KDD, &MODERATE_MASTER_KEY)).unwrap();
+        }
+        test::cleanup_wallet("wallet_service_create_wallet_works_for_moderate_key_derivation");
     }
 
     #[test]
+    #[ignore]
```

Comment:
Because it always fails on my machine and it is unclear whether the resolution of the system clock is high enough to detect the difference of nearly equal values. I think this test does not make sense, please explain if you think it does.